### PR TITLE
NEW Add option PDF_HIDE_PRODUCT_REF_IN_SUPPLIER_LINES

### PR DESCRIPTION
**Pull request by bafbes**
# New [\* Hide our ref in supplier orders by default and provide a parameter to show it*]

[\* The presentation of the supplier reference in supplier order documents is ugly and take too much place. As the supplier do not need to know our reference, its a bad idea to be obliged to show it. So we hide our ref  by default and provide a parameter to allow to show it*]

### CODE REVIEW

File: htdocs/core/lib/pdf.lib.php [hunk 1]
```diff
@@ -1077,7 +1077,7 @@ function pdf_writelinedesc(&$pdf,$object,$i,$outputlangs,$w,$h,$posx,$posy,$hide
  *  @param  int			$issupplierline      Is it a line for a supplier object ?
  *  @return string       				     String with line
  */
-function pdf_getlinedesc($object,$i,$outputlangs,$hideref=0,$hidedesc=0,$issupplierline=0)
+function pdf_getlinedesc($object,$i,$outputlangs,$hideref=0,$hidedesc=0,$issupplierline=0,$showrefcustomerinsupplierline=0)
```
Comment (eldy) [inline htdocs/core/lib/pdf.lib.php]: What is known into your local system is not the supplier ref. The supplier knows its ref, but yourself, you know your own ref. So you should see first your own ref. You can introduce an option, but can you invert it and name it $hideprodrefifsupplierline (with default value to 0)

Comment (bafbes) [inline htdocs/core/lib/pdf.lib.php]: The supplier order is normally printed prior to be sent to the supplier. There is generally no need to print this document for own use, whatever I provide the parameter to restore this behaviour. Perhaps should I provide also a database constant to set $showrefinsupplierline. I inverted the default behaviour as I disliked the  appearance of line labels in supplier orders. Don't you?

Perhaps, should I also rename $showrefcustomerinsupplierline to $showrefinsupplierline !

As this function is also used by other document types such as supplier invoices, perhaps I need to take as basis of my reasoning  those other documents !

**Comment by marcosgdf** [comment]
Companies that use supplier orders usually want their reference in the document… So that there is an easy way to trace down orders and invoices

**Comment by bafbes** [comment]
How can we hide our own reference to the supplier when we want to be able to hide this information. This is useful in order to avoid to the supplier the confusion between his references and ours and in the same time we would produce more lightweight order documents.

We can use a database constant to select this behaviour ?
Which behaviour must have more priority ?

**Comment by aljawaid** [comment]
I agree, currently I cannot use dolibarr supplier orders as the supplier is confused and said they cannot work with it. My reference for products are descriptive words but the supplier's reference is only numbers.  When I print or email the supplier order it should show the suppliers reference only. Otherwise its actually pointless at the moment.

Maybe have an extra link to a document which shows both references to match them but we still need one document which shows the order with the suppliers references.

**Comment by bafbes** [comment]
So aljawaid thinks that we must hide our reference in supplier orders by default.

**Comment by BebZ** [comment]
Hey !
Why you don't close the discussion by adding an option 
to display OR NOT to display our ref in supplier orders ? 
are we obliged to change the default behaviour ? 
and then have a dozen of opened useless bugs like : 
BUG : our ref doesn't display in supplier orders... 
because not everyone follow devs and changelogs...

**Comment by bafbes** [comment]
Added option to show our product reference in supplier orders
https://github.com/Dolibarr/dolibarr/commit/1612fcf54534e3c60fbd8a2cb42c6d65563b102a

2016-03-14 8:40 GMT+01:00 BebZ notifications@github.com:

**Comment by bafbes** [comment]
Conflicts solved

**Comment by eldy** [comment]
The option does not need to be a parameter of function.
We already provide data to say it is a supplier object ($issupplierline)

So using option $conf->global->MAIN_HIDE_SUPPLIER_REF to hide supplier ref must be introduce int pdf_getlinedesc if issupplierline is on

**Comment by bafbes** [comment]
While pdf_getlinedesc hides or shows the customer ref depending on the value of the parameter $hiderefcustomerinsupplierline with which it's called, the calling statement to this function reports  the value of the ORDERSUPPLIER_HIDEREFCUSTOMERINSUPPLIERLINE constant as a calling parameter.

**Comment by eldy** [comment]
And why using a different option than MAIN_HIDE_SUPPLIER_REF ?  Is distinction required ?

**Comment by bafbes** [comment]
The constant MAIN_HIDE_SUPPLIER_REF may be used as its name suggests to  the supplier reference in whatever document or page, while  ORDERSUPPLIER_HIDEREFCUSTOMERINSUPPLIERLINE suggests to hide this reference in supplier orders only to avoid confusion with any other documents. For example, there is no need to hide the supplier reference in supplier invoices as these documents are no intended to be sent outside the company.

**Comment by aljawaid** [comment]
Is there a way to show the supplier's reference instead of our reference?

**Pull request closed without merging**